libsolidity/analysis/StaticAnalyzer.h

```diff
@@ -85,6 +87,10 @@ class StaticAnalyzer: private ASTConstVisitor
 	/// when traversing.
 	std::map<std::pair<size_t, VariableDeclaration const*>, int> m_localVarUseCount;
 
+	/// Cache that holds information about whether a contract's constructor
+	/// uses inline assembly.
+	std::shared_ptr<ConstructorUsesAssembly> m_constructorUsesAssembly;
```

Comment:
```suggestion
	std::unique_ptr<ConstructorUsesAssembly> m_constructorUsesAssembly;
```
Unless I'm missing something, ownership is very clear here and I don't think ``StaticAnalyzer`` needs to be copyable.

Reply:
The problem is that the class needs to be fully known to use unique_ptr here.